gef.py

```diff
@@ -5221,186 +5412,43 @@ class PCustomCommand(GenericCommand):
 
     def __init__(self) -> None:
         super().__init__(prefix=True)
-        self["struct_path"] = (os.sep.join( (gef.config["gef.tempdir"], "structs")), "Path to store/load the structure ctypes files")
+        self["struct_path"] = (str(pathlib.Path(gef.config["gef.tempdir"]) / "structs"), "Path to store/load the structure ctypes files")
         self["max_depth"] = (4, "Maximum level of recursion supported")
         self["structure_name"] = ("bold blue", "Color of the structure name")
         self["structure_type"] = ("bold red", "Color of the attribute type")
         self["structure_size"] = ("green", "Color of the attribute size")
         return
 
-
-    def do_invoke(self, argv: List) -> None:
-        argc = len(argv)
-        if argc == 0:
+    @parse_arguments({"type": "", "address": ""}, {})
+    def do_invoke(self, *args: List, **kwargs: Dict[str, Any]) -> None:
+        args = kwargs["arguments"]
+        if not args.type:
             gdb.execute("pcustom list")
             return
 
-        modname, structname = self.get_modulename_structname_from_arg(argv[0])
+        _, structname = self.explode_type(args.type)
 
-        if argc == 1:
+        if args.type and not args.address:
             gdb.execute(f"pcustom show {structname}")
-        else:
-            try:
-                address = parse_address(argv[1])
-            except gdb.error:
-                err(f"Failed to parse '{argv[1]}'")
-                return
+            return
+
+        if not is_alive():
```

Comment:
Why not use `@only_if_gdb_running` instead? Is it just for 'auto' `pcustom list` support?

Reply:
Yes and `dt $structure`